shared/lib/routes/example/example.js

```diff
@@ -1,27 +1,30 @@
-/*global JS_ENV Promise require*/
+/*global JS_ENV Map Promise require*/
 
 import RouteBase from './../route.base';
 
 class ExampleRoute extends RouteBase {
 
-  get route_name(){
+  get route_name() {
     return 'ExampleRoute';
   }
 
-  assureData(state){
+  assureData(state) {
     let route = this;
-
-    if (!route.stateExampleSet(state)){
+    if (!this.repo) {
       let ExampleRepo = require(`./../../../../${JS_ENV}/models/example/example.repository.js`);
-      return ExampleRepo.findById(route.params.example_id)
-        .then((example)=>{
+      this.repo = new ExampleRepo(new Map());
+    }
+
+    if (!route.stateExampleSet(state)) {
+      return this.repo.findById(route.params.example_id)
+        .then((example) => {
           state.example = example;
         });
     }
     return Promise.resolve();
   }
 
-  stateExampleSet(state){
+  stateExampleSet(state) {
     return state.example && parseInt(this.params.example_id) === parseInt(state.example.data.id)
   }
 
```

Comment:
The one question I have about this is if we instantiate the repo with a new store in separate files (for instance if two separate routes used `ExampleRepo`), the data between those files will not be shared - so we'd have redundant data fetching (ie API calls) and caching.

If we avoid static methods, then we could `module.exports = new ExampleRepository(new Map());` in `example.repository.js`. 

What do you think?